public/app/features/alerting/unified/components/contact-points/useContactPoints.tsx

```diff
@@ -173,6 +209,66 @@ export const useGrafanaContactPoints = ({
   ]);
 };
 
+/**
+ * Fetch single contact point via the alertmanager config
+ */
+const useGetAlertmanagerContactPoint = (
+  { alertmanager, name }: BaseAlertmanagerArgs & { name: string },
+  queryOptions?: Parameters<typeof useGetAlertmanagerConfigurationQuery>[1]
+) => {
+  return useGetAlertmanagerConfigurationQuery(alertmanager, {
+    ...queryOptions,
+    selectFromResult: (result) => {
+      const matchedContactPoint = result.data?.alertmanager_config.receivers?.find(
+        (receiver) => decodeURIComponent(receiver.name) === decodeURIComponent(name)
```

Comment:
I'd say that decoding URIs shouldn't be responsibility of the API client. As an API consumer I'd be quite surprised to see that parameters I passed as options are altered in any way.
The only reason we need to do this is because we get the name parameter from URL
wdyt?

Reply:
Thanks, I've moved the decoding so its a bit cleaner now